backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailAggregator.java

```diff
@@ -60,42 +57,38 @@
-                                                                aggregationKey.getAccountId(),
-                                                                Stream.concat(
-                                                                        endpoints
-                                                                                .stream()
-                                                                                .map(EndpointRecipientSettings::new),
-                                                                        ActionRecipientSettings.fromAction(action).stream()
-                                                                ).collect(Collectors.toSet()),
-                                                                users
-                                                        )
-                                                )
-                                        )
-                                        /*
-                                         * We now have the final recipients list.
-                                         * Let's populate with synchronized side-effects (invoke) the Map that will be returned by the method.
-                                         */
-                                        .onItem().invoke(users -> {
-                                            users.forEach(user -> {
-                                                // It's aggregation time!
-                                                fillUsers(aggregationKey, user, aggregated, aggregation);
-                                            });
-                                        });
+                    // We need its event type to determine the target endpoints.
+                    String eventType = getEventType(aggregation);
+                    // Let's retrieve these targets.
+                    return endpointResources.getTargetEndpointsFromType(aggregationKey.getAccountId(), aggregationKey.getBundle(), aggregationKey.getApplication(), eventType, EndpointType.EMAIL_SUBSCRIPTION)
+                            .onItem().transform(Set::copyOf)
+                            // Now we want to determine who will actually receive the aggregation email.
+                            .onItem().transformToUni(endpoints ->
+                                    // All users who subscribed to the current application and subscription type combination are recipients candidates.
+                                    subscribers.onItem().transformToUni(users ->
+                                            /*
+                                             * The actual recipients list may differ from the candidates depending on the endpoint properties and the action settings.
+                                             * The target endpoints properties and the action settings will determine whether or not each candidate will actually receive an email.
+                                             */
```

Comment:
Shouldn't this comment be reverted as well since the recipients no longer come from the action?